packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -1028,10 +1039,15 @@ export function convertCookiesExpiryCdpToBiDi(
 }
 
 export function convertCookiesPartitionKeyFromPuppeteerToBiDi(
-  partitionKey: ChromeCookiePartitionKey | string | undefined,
+  partitionKey: CookiePartitionKey | string | undefined,
 ): string | undefined {
   if (partitionKey === undefined || typeof partitionKey === 'string') {
     return partitionKey;
   }
-  return partitionKey.topLevelSite;
+  if (partitionKey.hasCrossSiteAncestor) {
+    throw new UnsupportedOperation(
+      'WebDriver BiDi does not support goog:hasCrossSiteAncestor yet.',
```

Comment:
I think we don't want to mention the prefix
```suggestion
      'WebDriver BiDi does not support `hasCrossSiteAncestor` yet.',
```